src/Symfony/Bundle/FrameworkBundle/Tests/EventListener/ResolveControllerNameSubscriberTest.php

```diff
@@ -51,4 +51,23 @@ public function testSkipsOtherControllerFormats()
         $subscriber->onKernelRequest(new GetResponseEvent($httpKernel, $request, HttpKernelInterface::MASTER_REQUEST));
         $this->assertEquals('Other:format', $request->attributes->get('_controller'));
     }
+
+    public function testSkipsNonStringControllers()
```

Comment:
@alekitto, you could use the [data provider](https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers) to avoid a code duplication. Like this:
```php
/** @dataProvider provideSkippedControllers */
public function testSkipsOtherControllerFormats($controller)
{
    // ...
    $this->assertEquals($controller, $request->attributes->get('_controller'));
}

private function provideSkippedControllers()
{
    yield [ 'Other:format' ];
    yield [ function() {} ];
}
```